File: 102. Binary Tree Level Order Traversal/other2.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @description https://leetcode.com/problems/binary-tree-level-order-traversal/
 * @tutorial https://leetcode-cn.com/problems/binary-tree-level-order-traversal/solution/er-cha-shu-de-ceng-xu-bian-li-by-leetcode-solution/
 * @param {TreeNode} root
 * @return {number[][]}
 */
var levelOrder = function (root) {
  if (!root) {
    return [];
  }
  var levels = [];

  function helper(node, level) {
    if (levels.length === level) {
      levels.push([]);
    }
    levels[level].push(node.val);
    if (node.left) {
      helper(node.left, level + 1);
    }
    if (node.right) {
      helper(node.right, level + 1);
    }
  }

  helper(root, 0);

  return levels;
};

File: 1315. Sum of Nodes with Even-Valued Grandparent/other2.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
/** 
 * https://leetcode.com/problems/sum-of-nodes-with-even-valued-grandparent/
 * https://leetcode.com/problems/sum-of-nodes-with-even-valued-grandparent/discuss/477048/JavaC%2B%2BPython-1-Line-Recursive-Solution @nice_dev
 */
class Solution {
  public int sumEvenGrandparent(TreeNode root) {
    if(root == null) return 0;
    int sum = 0;
    if((root.val % 2) == 0){
      if(root.left != null && root.left.left != null) sum += root.left.left.val;
      if(root.left != null && root.left.right != null) sum += root.left.right.val;
      if(root.right != null && root.right.left != null) sum += root.right.left.val;
      if(root.right != null && root.right.right != null) sum += root.right.right.val;
    }
    return sum + sumEvenGrandparent(root.left) + sumEvenGrandparent(root.right);
  }
}